src/uvw/resource.hpp

```diff
@@ -29,19 +29,19 @@ class Resource: public Emitter<T>, public std::enable_shared_from_this<T> {
           pLoop{std::move(ref)},
           resource{}
     {
-        this->template get<U>()->data = static_cast<T*>(this);
+        resource.data = static_cast<T*>(this);
     }
 
     auto parent() const noexcept {
         return pLoop->loop.get();
     }
 
-    template<typename R>
+    template<typename R = U>
```

Comment:
Give me a while to think about it.
I don't like it too much, for we are _hiding_ the type during the call to `get` and I usually like this kind of things to be explicit.
Moreover, in terms of code bloating, there is no benefit, for you are not removing a template specialization.
Anyway, I see your point, even though I'm not sure it's really an improvement in terms of readability.